go/vt/vtgate/engine/concatenate.go

```diff
@@ -151,10 +155,10 @@ func (c *Concatenate) execSources(vcursor VCursor, bindVars map[string]*querypb.
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			result, err := vcursor.ExecutePrimitive(currSource, vars, wantfields)
+			result, err := vcursor.ExecutePrimitive(ctx, currSource, vars, wantfields)
 			if err != nil {
 				outerErr = err
-				vcursor.CancelContext()
+				cancel()
```

Comment:
This would change the behavior too. We are now only canceling the context that we have created and not the vcursor context